extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+			}
+		} else {
+			$localHash=substr(fread($file,40),8);
+			fclose($file);
+			if ($hash!=$localHash) {
+				return false; // Wrong hash
+			}
+
+			# Require overwrites the member variable or just shadows it?
+			require( $filename );
+			return $this->setCache( $this->mCache, $code );
+		}
+	}
+
+	/**
+	 * Save the cache to a local file.
+	 */
+	function saveToLocal( $serialized, $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$dirname = $this->mLocalMessageCache . '/' . substr( $wgDefaultMessagesDB, 0, 1 ) . '/' . $wgDefaultMessagesDB;
+		$filename = "$dirname/messages-$code";
+		wfMkdirParents( $dirname, 0777 ); // might fail
+
+		wfSuppressWarnings();
+		$file = fopen( $filename, 'w' );
+		wfRestoreWarnings();
+
+		if ( !$file ) {
+			wfDebug( "Unable to open local cache file for writing\n" );
+			return;
+		}
+
+		fwrite( $file, $hash . $serialized );
+		fclose( $file );
+		@chmod( $filename, 0666 );
+	}
+
+	function saveToScript( $array, $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$filename = $this->mLocalMessageCache . "/messages-$wgDefaultMessagesDB-$code";
+		$tempFilename = $filename . '.tmp';
+		wfMkdirParents( $this->mLocalMessageCache, 0777 ); // might fail
+
+		wfSuppressWarnings();
+		$file = fopen( $tempFilename, 'w');
+		wfRestoreWarnings();
+
+		if ( !$file ) {
+			wfDebug( "Unable to open local cache file for writing\n" );
+			return;
+		}
+
+		fwrite($file,"<?php\n//$hash\n\n \$this->mCache = array(");
+
+		foreach ($array as $key => $message) {
+			$key = $this->escapeForScript($key);
+			$messages = $this->escapeForScript($message);
+			fwrite($file, "'$key' => '$message',\n");
+		}
+
+		fwrite($file,");\n?>");
+		fclose($file);
+		rename($tempFilename, $filename);
+	}
+
+	function escapeForScript($string) {
+		$string = str_replace( '\\', '\\\\', $string );
+		$string = str_replace( '\'', '\\\'', $string );
+		return $string;
+	}
+
+	/**
+	 * Set the cache to $cache, if it is valid. Otherwise set the cache to false.
+	 */
+	function setCache( $cache, $code ) {
+		if ( isset( $cache['VERSION'] ) && $cache['VERSION'] == MSG_CACHE_VERSION ) {
+			$this->mCache[$code] = $cache;
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Loads messages from caches or from database in this order:
+	 * (1) local message cache (if $wgLocalMessageCache is enabled)
+	 * (2) memcached
+	 * (3) from the database.
+	 *
+	 * When succesfully loading from (2) or (3), all higher level caches are
+	 * updated for the newest version.
+	 *
+	 * Nothing is loaded if  member variable mDisabled is true, either manually
+	 * set by calling code or if message loading fails (is this possible?).
+	 *
+	 * Returns true if cache is already populated or it was succesfully populated,
+	 * or false if populating empty cache fails. Also returns true if MessageCache
+	 * is disabled.
+	 *
+	 * @param $code String: language to which load messages
+	 */
+	function load( $code = false, $useDB = true ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)